storm-client/src/jvm/org/apache/storm/messaging/netty/BackPressureStatus.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+package org.apache.storm.messaging.netty;
+
+import org.apache.storm.serialization.KryoValuesDeserializer;
+import org.apache.storm.serialization.KryoValuesSerializer;
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.buffer.ChannelBuffers;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicLong;
+
+// Instances of this type are sent from NettyWorker to upstream WorkerTransfer to indicate BackPressure situation
+public class BackPressureStatus {
+    public static final short IDENTIFIER = (short)-600;
+    private static final int SIZE_OF_ID = 2; // size if IDENTIFIER
+    private static final int SIZE_OF_INT = 4;
+
+    private static AtomicLong bpCount = new AtomicLong(0);
+
+    public String workerId;
+    public final long id;                       // monotonically increasing id
```

Comment:
Sorry to leave two duplicated comments: I commented to old commit.

Just wondering: the characteristic of `monotonically increasing` guarantee `id` to be unique in a worker, but not among workers, and also reset to 0 after worker crash and restart. Does it hurt the backpressure logic at any chance?

Reply:
Its only for debugging purposes.. so that we can co-relate sent & recvd msgs. I have used it to measure latency involved in transmission of BackPressureStatus msgs.

Reply:
OK thanks for clarification. I thought it as some kinds of guarantee we should ensure. Maybe better to clear out that that's not a requirement and only for debugging purpose.